### PR TITLE
Products Table

### LINKED ISSUE #4: [Umbrella] Create Database Entities
image

### First we are going to create the classes
- [x] #7
- [x] #8
- [x] #9
- [x] #10
- [x] #11

### LINKED ISSUE #5: Product Repository
## After we create all the entities and have in mind all the table relationship we can discuss the projects repository functions

### LINKED ISSUE #11: Products

### LINKED ISSUE #22: /api/products

### LINKED ISSUE #23: /api/products/id

### PR DESCRIPTION
**Pull request by tatacsd**
Closes #11

### CODE REVIEW

File: PCU/src/main/java/com/coffeelovers/pcu/model/Product.java [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+	@GeneratedValue(strategy = GenerationType.AUTO)
+	private long productID;
+	
+	@Column(name= "materialsID")
+	private long MaterialsID;
+	
+	@Column(name= "variableCosts")
+	private long variableCosts;
+	
+	@Column(name= "pic")
```
Comment (tatacsd) [inline PCU/src/main/java/com/coffeelovers/pcu/model/Product.java:24]: @Andrew-shanan I was thinking about adding a more descriptive name for the column.

File: PCU/src/main/java/com/coffeelovers/pcu/model/Product.java [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+	private long pic;
+	
+	@Column(name= "size")
+	private long size;
```
Comment (tatacsd) [inline PCU/src/main/java/com/coffeelovers/pcu/model/Product.java:28]: @Andrew-shanan, what are we thinking about a size format?

File: PCU/src/main/java/com/coffeelovers/pcu/model/Product.java [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+	@Column(name= "size")
+	private String size;
+	
+	public Product() {
+		
+	}
+
+	// Personalized
+	public Product(long materialsID, double variableCosts, long pic, String size) {
+		this.materialsID = materialsID;
```
Comment (tatacsd) [inline PCU/src/main/java/com/coffeelovers/pcu/model/Product.java:36]:
```suggestion
		this.materialsID = materialID;
```

File: PCU/src/main/java/com/coffeelovers/pcu/model/Product.java [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+
+import javax.persistence.Column;
+import javax.persistence.Entity;
+import javax.persistence.GeneratedValue;
+import javax.persistence.GenerationType;
+import javax.persistence.Id;
+import javax.persistence.Table;
+
+@Entity
+@Table(name="products")
```
Comment (Andrew-shanan) [inline PCU/src/main/java/com/coffeelovers/pcu/model/Product.java]: the name of the table should be without "s"
```suggestion
@Table(name="product")
```

File: PCU/src/main/java/com/coffeelovers/pcu/model/Product.java [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+	@GeneratedValue(strategy = GenerationType.AUTO)
+	private long productID;
+	
+	@Column(name= "materialsID")
+	private long materialsID;
+	
+	@Column(name= "variableCosts")
+	private double variableCosts;
+	
+	@Column(name= "pic")
```
Comment (Andrew-shanan) [inline PCU/src/main/java/com/coffeelovers/pcu/model/Product.java]: please add full name "Picture"
```suggestion
	@Column(name= "Picture")
```

File: PCU/src/main/java/com/coffeelovers/pcu/model/ProductRepository.java [hunk 6]
```diff
@@ -0,0 +1,12 @@
+package com.coffeelovers.pcu.model;
+
+import java.util.List;
+import org.springframework.data.jpa.repository.JpaRepository;
+
+public interface ProductRepository extends JpaRepository<Product, Long> {
+	// Add the abstract methods
```
Comment (tatacsd) [inline PCU/src/main/java/com/coffeelovers/pcu/model/ProductRepository.java]:
```suggestion
```

**Review by Andrew-shanan — APPROVED** [review]

**Pull request merged**